lib/apiservers/engine/backends/container.go

```diff
@@ -671,13 +673,13 @@ func (c *Container) ContainerAttach(prefixOrName string, ca *backend.ContainerAt
 	vc := cache.ContainerCache().GetContainer(prefixOrName)
 
 	if vc == nil {
-		//FIXME: If we didn't find in the cache, we should goto the port layer and
-		//see if it exists there.  API server might have been bounced.  For now,
-		//just return error
-
 		return derr.NewRequestNotFoundError(fmt.Errorf("No such container: %s", prefixOrName))
 	}
 
+	if running, err := IsContainerRunning(vc.ContainerID); !running {
```

Comment:
This seems ripe for racing.  This is a test/set problem.  You might want to at a minimum do this in a loop for some duration of time (`>= ` time to start a container) before deciding if the container is indeed not running.

Is there not a subscription mechanism we can use here instead of polling?  That could be a later optimization but would be the preferable implementation.